### PR TITLE
ST-399: Change shipping/billing address

### PR DESCRIPTION
**Pull request by ilyawzrd**
Description:

--

QA-test: ST-401

Demo-test: ST-402

Download artifact URL: https://vc3prerelease.blob.core.windows.net/packages/client-app-0.0.1-pr-45-c7678a47.zip

### CODE REVIEW

File: client-app/shared/account/composables/useUser.ts [hunk 1]
```diff
@@ -1,9 +1,16 @@
 import { Ref, ref, computed } from "vue";
-import { getMe, updatePersonalData } from "@/core/api/graphql/account";
-import { UserType, IdentityResultType } from "@core/api/graphql/types";
+import {
+  getMe,
+  getMyAddresses,
+  updateMemberAddresses,
+  updatePersonalData,
+  getDefaultShippingAddress,
```
Comment (artmakarov) [inline client-app/shared/account/composables/useUser.ts:7]: it seems that some of this is not being used

File: client-app/shared/cart/composables/useCart.ts [hunk 2]
```diff
@@ -11,7 +11,7 @@ import {
   addOrUpdateCartShipment,
   addOrUpdateCartPayment,
 } from "@core/api/graphql/cart";
-import { CartType, InputPaymentType, InputShipmentType, LineItemType } from "@core/api/graphql/types";
+import { CartType, InputAddressType, InputPaymentType, InputShipmentType, LineItemType } from "@core/api/graphql/types";
```
Comment (artmakarov) [inline client-app/shared/cart/composables/useCart.ts:14]: is it used anywhere?

File: client-app/shared/cart/composables/useCheckout.ts [hunk 3]
```diff
@@ -17,6 +17,7 @@ import {
   ShipmentType,
   PaymentType,
   CartType,
+  MemberAddressType,
```
Comment (artmakarov) [inline client-app/shared/cart/composables/useCheckout.ts:20]: is it used anywhere?

**Review by Flanker72 — APPROVED** [review]

**Review by artmakarov — APPROVED** [review]

File: client-app/pages/checkout/checkout.vue [hunk 4]
```diff
@@ -204,14 +271,28 @@ const cartCoupon = ref("");
...
 const cartComment = ref("");
 
-const isValidCheckout = computed(() => !(cart.value.validationErrors && cart.value.validationErrors?.length > 0));
+const isValidCheckout = computed(
+  () =>
+    !(cart.value.validationErrors && cart.value.validationErrors?.length > 0) &&
+    isValidShipment.value &&
+    isValidPayment.value
+);
+const isValidShipment = computed(
```
Comment (trueboroda) [inline client-app/pages/checkout/checkout.vue:317]: Add empty line

File: client-app/pages/checkout/checkout.vue [hunk 5]
```diff
@@ -204,14 +271,28 @@ const cartCoupon = ref("");
...
+  () =>
+    cart.value.shipments?.[0] &&
+    cart.value.shipments?.[0]?.shipmentMethodCode &&
+    cart.value.shipments?.[0]?.deliveryAddress
+);
+const isValidPayment = computed(() => cart.value.payments?.[0] && cart.value.payments?.[0]?.paymentGatewayCode);
```
Comment (trueboroda) [inline client-app/pages/checkout/checkout.vue:323]: add empty line

File: client-app/pages/checkout/checkout.vue [hunk 6]
```diff
@@ -262,6 +343,12 @@ const createOrder = async () => {
     if (cartComment.value) {
       await changeComment(cartComment.value);
     }
+    if (billingSameAsShipping.value) {
```
Comment (trueboroda) [inline client-app/pages/checkout/checkout.vue:374]: empty line

File: client-app/pages/checkout/checkout.vue [hunk 7]
```diff
@@ -262,6 +343,12 @@ const createOrder = async () => {
...
+      await updatePayment({
+        id: cart.value.payments?.[0]?.id,
+        billingAddress: { ...cart.value.shipments?.[0]?.deliveryAddress },
+      });
+    }
     await placeOrder(cart.value.id).then((order) => {
```
Comment (trueboroda) [inline client-app/pages/checkout/checkout.vue:380]: empty...

File: client-app/shared/checkout/components/shipping-address-dialog.vue [hunk 8]
```diff
@@ -0,0 +1,158 @@
...
+const selectedAddress = ref();
+
+function setAddress(address: MemberAddressType): void {
+  selectedAddress.value = address;
+}
+
+onMounted(async () => {
+  const result = await getMyAddresses();
+  availableAddresses.value = result;
+  selectedAddress.value = availableAddresses.value.find((item) => {
```
Comment (trueboroda) [inline client-app/shared/checkout/components/shipping-address-dialog.vue:151]: empty line

File: client-app/shared/checkout/components/shipping-address-dialog.vue [hunk 9]
```diff
@@ -0,0 +1,158 @@
...
+
+const breakpoints = useBreakpoints(breakpointsTailwind);
+const isMobile = breakpoints.smaller("md");
+
+const availableAddresses = ref<MemberAddressType[]>([]);
+
+const page = ref(1);
+const itemsPerPage = ref(4);
+const pages = computed(() => Math.ceil(availableAddresses.value.length / itemsPerPage.value));
+const paginatedAddresses = computed(() =>
```
Comment (trueboroda) [inline client-app/shared/checkout/components/shipping-address-dialog.vue:139]: empty line

File: client-app/shared/checkout/components/shipping-address-dialog.vue [hunk 10]
```diff
@@ -0,0 +1,158 @@
...
+  availableAddresses.value.slice((page.value - 1) * itemsPerPage.value, page.value * itemsPerPage.value)
+);
+const selectedAddress = ref();
```
Comment (trueboroda) [inline client-app/shared/checkout/components/shipping-address-dialog.vue:142]: empty empty line

**Review by trueboroda — APPROVED** [review]

**Pull request merged**